tests/unit/test_fm.py

```diff
@@ -94,3 +108,282 @@ def test_all_hyperparameters(sagemaker_session):
 def test_image(sagemaker_session):
     fm = FactorizationMachines(sagemaker_session=sagemaker_session, **ALL_REQ_ARGS)
     assert fm.train_image() == registry(REGION) + '/factorization-machines:1'
+
+
+def test_num_factors_validation_fail_type(sagemaker_session):
```

Comment:
I'd recommend using parameterized test functions to make these tests significantly more concise for each: https://docs.pytest.org/en/latest/parametrize.html

We can chat about the best way to do this if you like.

Reply:
As discussed offline, all hyper-parameter related unit tests are now in parametrized style.